src/Playwright/Core/Locator.cs

```diff
@@ -47,17 +48,25 @@ public Locator(Frame parent, string selector)
         public ILocator Last => new Locator(_frame, $"{_selector} >> _nth=last");
 
         public async Task<IReadOnlyList<string>> AllInnerTextsAsync()
-            => (await _frame.QuerySelectorAllAsync(_selector).ConfigureAwait(false))
-            .Select(async x => await x.InnerTextAsync().ConfigureAwait(false))
-            .Where(x => x.Result != null)
-            .Select(x => x.Result)
-            .ToArray();
+        {
+            var handles = await _frame.QuerySelectorAllAsync(_selector).ConfigureAwait(false);
+            return (await handles
+                    .Select(async x => await x.InnerTextAsync().ConfigureAwait(false))
+                    .WaitForResultsAsync()
+                    .ConfigureAwait(false))
+                .ToArray();
+        }
```

Comment:
```suggestion
        => (await (await _frame.QuerySelectorAllAsync(_selector).ConfigureAwait(false))
            .SelectAsync(async x => await x.InnerTextAsync().ConfigureAwait(false)).ConfigureAwait(false))
            .Where(x => !string.IsNullOrEmpty(x))
            .ToArray();
```

Reply:
Thanks @avodovnik, I've update this PR with your `SelectAsync` suggested change.
I deliberately refactored to avoid `(await (await ` as I found it a bit disorientating personally. Let me know if you disagree and I'll apply the suggestion here 🙂